Add CLI_SETUP and delete BYPASS_CLI_BUILD (#2450)
hashicorp/boundary-ui: ui/desktop/README.md

- `yarn start:desktop`

> The Boundary CLI is NOT downloaded by default, to download and extract the CLI to `electron-app/cli` folder as part of the build, you need to set environment variable `CLI_SETUP` to true. Example: `CLI_SETUP=true yarn start:desktop`.

Reviewer: ```suggestion
The Boundary CLI is NOT downloaded by default, to download and extract the CLI to `electron-app/cli` folder as part of the build, you need to set the environment variable `CLI_SETUP` to true. Example: `CLI_SETUP=true yarn start:desktop`.
```

> CLI version is defined in `electron-app/config.cli.js`.

Reviewer: ```suggestion
The CLI version is defined in `electron-app/config/cli/VERSION`.
```
Don't we define it in the `VERSION` file?

Author: You are correct, I made the confusion here on wording. When I mean the version is defined within config.cli.js I mean the script will generate what version to download via the url, but you are totally correct, the CLI version is defined within the VERSION file. Accepting your changes 😉 

Thanks for the catch!

### Developing Using Non-Release Versions of Boundary

> To develop using a non-release version of Boundary, download the Boundary CLI version you want to use and extract it at `electron-app/cli` folder. You may need to create the directory or to cleaning it up beforehand.

Reviewer: ```suggestion
To develop using a non-release version of Boundary, download the Boundary CLI version you want to use and extract it to the `electron-app/cli` folder. You may need to create the directory or clean it up beforehand.
```

| `APP_UPDATER_CURRENT_VERSION` | Version of client. |
| `APP_UPDATER_LATEST_VERSION_TAG` | Next version for comparison with current version. |
| `APP_UPDATER_LATEST_VERSION_LOCATION` | Location of app release to use for updating client. Can be a filepath or url. |

> | `CLI_SETUP` | Enable download and extraction of CLI. For development use only. |

Reviewer: Do you think this should be called `SETUP_CLI` instead since it's nicer to read with the verb in the beginning?

Is it still considered `"For development use only."` or should we remove that now?

Author: Absolutely agree. `SETUP_CLI` sounds and read way better. I will provide the change.

Yes!!! God catch, I will delete the "for dev only"
